File: ruby/publish-example.rb
require 'Pubnub'

publish_key   = ARGV[0]
subscribe_key = ARGV[1]
secret_key    = ARGV[2]
message       = ARGV[3]
ssl_on        = !!ARGV[4]

## Print usage if missing info.
if !message
    puts('
    Get API Keys at http://www.pubnub.com/account
    ==============
    EXAMPLE USAGE:
    ==============
    ruby publish-example.rb PUB-KEY SUB-KEY SECRET-KEY "message text" SSL-ON
    ruby publish-example.rb demo demo "" "hey what is up?" true

    ')
    exit()
end

## -----------------------------------------
## Create Pubnub Client API (INITIALIZATION)
## -----------------------------------------
puts('Creating new Pubnub Client API')
pubnub = Pubnub.new( publish_key, subscribe_key, secret_key, ssl_on )

## ----------------------
## Send Message (PUBLISH)
## ----------------------
puts('Sending a message with publish() Function')
info = pubnub.publish({
    'channel' => 'hello_world',
    'message' => message
})

## Print Pretty
puts(info)

File: php/3.2/examples/publish-example.php
<?php

    ## Capture Publish and Subscribe Keys from Command Line
    $publish_key   = isset($argv[1]) ? $argv[1] : false;
    $subscribe_key = isset($argv[2]) ? $argv[2] : false;
    $message       = isset($argv[3]) ? $argv[3] : false;
    $channel_name  = isset($argv[4]) ? $argv[4] : 'hello_world2';

    ## Print usage if missing info.
    if (!($publish_key && $subscribe_key && $message)) {
echo("
    ==============
    EXAMPLE USAGE:
    ==============
    php ./publish-example.php PUBLISH-KEY SUBSCRIBE-KEY 'ANY MESSAGE HERE' 'CHANNEL_NAME'
    php ./publish-example.php demo demo 'Hey what is up?' 'hello_world_channel'

");
        exit();
    }

    ## Require Pubnub API
    echo("Loading Pubnub.php Class\n");
    require('../Pubnub.php');

    ## -----------------------------------------
    ## Create Pubnub Client API (INITIALIZATION)
    ## -----------------------------------------
    echo("Creating new Pubnub Client API\n");
    $pubnub = new Pubnub( $publish_key, $subscribe_key );

    ## ----------------------
    ## Send Message (PUBLISH)
    ## ----------------------
    echo("Sending a message with Publish Function\n");
    $message = array( 'text' => $message );
    $info = $pubnub->publish(array(
        'channel' => $channel_name,
        'message' => $message
    ));

    echo("Response:\n");
    print_r(array( 'published' => $message/*, 'success' => $info*/ ));

?>
